flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -825,6 +841,38 @@ public <N, SV, SEV, S extends State, IS extends S> IS createInternalState(
             @Nonnull StateDescriptor<S, SV> stateDesc,
             @Nonnull StateSnapshotTransformFactory<SEV> snapshotTransformFactory)
             throws Exception {
+        Tuple2<ColumnFamilyHandle, RegisteredKeyValueStateBackendMetaInfo<N, SV>> registerResult =
+                tryRegisterKvStateInformation(
+                        stateDesc, namespaceSerializer, snapshotTransformFactory, false);
+        ttlCompactFiltersManager.configCompactFilter(
+                stateDesc, registerResult.f1.getStateSerializer());
+
+        return createState(stateDesc, registerResult);
```

Comment:
Currently, this class implements all `KeyedStateFactory.createInternalState` overloaded alternatives, which adds duplication and complexity.
Is it possible to call one `createInternalState()` from the other, passing some default arguments (e.g. `allowFutureMetadataUpdates=false`)?

Reply:
Currently I haven't done like this just because [above comment](https://github.com/apache/flink/pull/19679#discussion_r935732150).

Reply:
I think this might cause issues in the future, if a developer expects the same behavior from the two methods; but the 2nd won't actually apply TTL settings.

To solve this, the 2nd version could call `ttlCompactFiltersManager.configCompactFilter` if `allowFutureMetadataUpdates == true`. 
That might seem hacky, but it conveys the current logic better IMO, which is:
only take TTL config into account if no future metadata updates (and hence, TTL) will happen.

WDYT?

Reply:
I think this might cause issues in the future, if a developer expects the same behavior from the two methods; but the 2nd won't actually apply TTL settings.

To solve this, the 2nd version could call `ttlCompactFiltersManager.configCompactFilter` if `allowFutureMetadataUpdates == true`. 
That might seem hacky, but it conveys the current logic better IMO, which is:
only take TTL config into account if no future metadata updates (and hence, TTL) will happen.

WDYT?

Reply:
It looks like clear, I have modified and added comment.